static/js/protein/chart.ts

```diff
@@ -432,6 +460,280 @@ export function drawProteinInteractionChart(
+
+  const nodeLabels = nodes
+    .append("text")
+    .attr("fill", NODE_STYLE.labels.font.color)
+    .attr("text-anchor", "middle")
+    .attr("dominant-baseline", "middle")
+    .style(
+      "font",
+      `${NODE_STYLE.labels.font.size} ${NODE_STYLE.labels.font.name}`
+    )
+    .text(({ name }) => `${name}`);
+
+  // floating name labels we can switch in
+  // const labels = node
+  //   .append("text")
+  //   .attr("fill", "#555")
+  //   .attr("dx", 15)
+  //   .attr("dy", -15)
+  //   .text(({name}) => `${name}`)
+
+  function ticked(): void {
+    // update node and link positions
+
+    // type assertions needed because x,y info added after initialization
+    // https://github.com/tomwanzek/d3-v4-definitelytyped/blob/06ceb1a93584083475ecb4fc8b3144f34bac6d76/src/d3-force/index.d.ts#L24
+    links
+      .attr(
+        "x1",
+        (linkSimulationDatum) =>
+          (linkSimulationDatum.source as SimulationNodeDatum).x
+      )
+      .attr(
+        "y1",
+        (linkSimulationDatum) =>
+          (linkSimulationDatum.source as SimulationNodeDatum).y
+      )
+      .attr(
+        "x2",
+        (linkSimulationDatum) =>
+          (linkSimulationDatum.target as SimulationNodeDatum).x
+      )
+      .attr(
+        "y2",
+        (linkSimulationDatum) =>
+          (linkSimulationDatum.target as SimulationNodeDatum).y
+      );
+
+    // same here
+    // https://github.com/tomwanzek/d3-v4-definitelytyped/blob/06ceb1a93584083475ecb4fc8b3144f34bac6d76/src/d3-force/index.d.ts#L13
+    nodes.attr(
+      "transform",
+      (nodeSimulationDatum) =>
+        `translate(${(nodeSimulationDatum as SimulationNodeDatum).x}, ${
+          (nodeSimulationDatum as SimulationNodeDatum).y
+        })`
+    );
+  }
+
+  function drag(
+    simulation: Simulation<ProteinNode, InteractionLink>
+  ): DragBehavior<Element, SimulationNodeDatum, SimulationNodeDatum> {
+    // Reference for alphaTarget: https://stamen.com/forcing-functions-inside-d3-v4-forces-and-layout-transitions-f3e89ee02d12/
+
+    function dragstarted(nodeDatum: ProteinNode): void {
+      if (!d3.event.active) {
+        simulation.alphaTarget(0.3).restart();
+      } // start up simulation
+      nodeDatum.fx = nodeDatum.x;
+      nodeDatum.fy = nodeDatum.y;
+    }
+
+    function dragged(nodeDatum: ProteinNode): void {
+      nodeDatum.fx = d3.event.x;
+      nodeDatum.fy = d3.event.y;
+    }
+
+    function dragended(nodeDatum: ProteinNode): void {
+      if (!d3.event.active) {
+        simulation.alphaTarget(0);
+      } // cool down simulation
+      nodeDatum.fx = null;
+      nodeDatum.fy = null;
+    }
+
+    return d3
+      .drag()
+      .on("start", dragstarted)
+      .on("drag", dragged)
+      .on("end", dragended);
+  }
+
+  // @reviewers: lightenThis/darkenThis should probably use color.darker() and color.brighter() instead so as not to alter transparency.
+  // However, transparency effect may be useful to view edges entering a node in a dense graph.
```

Comment:
if this is trying to show mouse hover state, I think changing the color would be better than opacity. These functions should also be defined outside of `drawProteinInteractionGraph` function.

When you have something you want to bring to the attention of reviewers, please just include a comment in your PR instead of including it in your code. Please remove these comments from the actual code.

Reply:
I spent like 2 hrs trying to figure this one out! Eventually went with [brightness](https://stackoverflow.com/a/69610045) which doesn't require stateful tracking of the original color of object under mouse and also doesn't alter transparency like opacity does.